### PR TITLE
Implement IsInf JIT Emitter for ARM64 SIMD in OpenVINO

### PR DESCRIPTION
**Pull request by inbasperu**
Hello maintainers,

I've implemented the IsInf JIT emitter for the ARM64 SIMD platform, as outlined in the OpenVINO CPU plugin JIT emitters documentation.

This PR addresses [#24419](https://github.com/openvinotoolkit/openvino/issues/24419).

Please let me know if any further adjustments are needed.
Thank you!

### CODE REVIEW

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 1]
```diff
@@ -452,6 +452,96 @@ std::set<std::vector<element::Type>> jit_hswish_emitter::get_supported_precision
...
+        }
+        // Compare elements of 'src' with 'aux'.
+        h->fcmeq(dst.s, src.s, aux.s);
+        // Sets elements in 'dst' to 1.0 where the comparison was true.
+        h->ld1r(aux.s, table_val2("one"));
+        h->and_(dst.b16, dst.b16, aux.b16);
+
+    } else {
+        // If neither positive nor negative infinity detection is enabled, load 'dst' with zeros.
+        h->ld1r(dst.s, table_val2("zero"));
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:533]: you don't need store `zero`, use `eor` instruction instead

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp [hunk 2]
```diff
@@ -187,6 +187,40 @@ class jit_hswish_emitter : public jit_emitter {
     void emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const;
 };
 
+class jit_is_inf_emitter : public jit_emitter {
+public:
+    jit_is_inf_emitter(dnnl::impl::cpu::aarch64::jit_generator* host,
+                       dnnl::impl::cpu::aarch64::cpu_isa_t host_isa,
+                       const ov::element::Type exec_prc = ov::element::f32,
+                       bool detect_negative = true,
+                       bool detect_positive = true);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp:196]:
```
const bool detect_negative = true,
const bool detect_positive = true);
```

**Review by eshoguli — COMMENTED** [review]
Thanks, good start, there are minor comments.
Don't forget about the tests, please.

**Comment by inbasperu** [comment]
> Thanks, good start, there are minor comments. Don't forget about the tests, please.

Thank you @eshoguli for the comments. I will make these changes. 
I had some questions about the tests. In the description of the issue [here](https://github.com/openvinotoolkit/openvino/issues/24419#issue-2285497080), it is mentioned that we should add tests for the supported operands in the respective [element-wise](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/eltwise.cpp#L257) or [activation](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/activation.cpp#L155) files. However, I am not sure how to proceed with the `ComparisonTypes`. I did look at a similar example, the equal JIT emitter, in [this pull request](https://github.com/openvinotoolkit/openvino/pull/23266), but couldn't find how to add tests. Could you please help me with this?

**Comment by eshoguli** [comment]
> > Thanks, good start, there are minor comments. Don't forget about the tests, please.
> 
> Thank you @eshoguli for the comments. I will make these changes. I had some questions about the tests. In the description of the issue [here](https://github.com/openvinotoolkit/openvino/issues/24419#issue-2285497080), it is mentioned that we should add tests for the supported operands in the respective [element-wise](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/eltwise.cpp#L257) or [activation](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/activation.cpp#L155) files. However, I am not sure how to proceed with the `ComparisonTypes`. I did look at a similar example, the equal JIT emitter, in [this pull request](https://github.com/openvinotoolkit/openvino/pull/23266), but couldn't find how to add tests. Could you please help me with this?

Use this commit https://github.com/eshoguli/openvino/commit/1e27b762d225ec02d920e02692ef4255db29239d for `Maximum` & `Minimum` operations as example how to easilly add new operation to existing eltwise test. Let me know if you still have any questions.

**Comment by inbasperu** [comment]
> > > Thanks, good start, there are minor comments. Don't forget about the tests, please.
> > 
> > 
> > Thank you @eshoguli for the comments. I will make these changes. I had some questions about the tests. In the description of the issue [here](https://github.com/openvinotoolkit/openvino/issues/24419#issue-2285497080), it is mentioned that we should add tests for the supported operands in the respective [element-wise](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/eltwise.cpp#L257) or [activation](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/activation.cpp#L155) files. However, I am not sure how to proceed with the `ComparisonTypes`. I did look at a similar example, the equal JIT emitter, in [this pull request](https://github.com/openvinotoolkit/openvino/pull/23266), but couldn't find how to add tests. Could you please help me with this?
> 
> Use this commit [eshoguli@1e27b76](https://github.com/eshoguli/openvino/commit/1e27b762d225ec02d920e02692ef4255db29239d) for `Maximum` & `Minimum` operations as example how to easilly add new operation to existing eltwise test. Let me know if you still have any questions.

Hey @eshoguli,
I have incorporated the code review feedback and added the IsInf operation to the existing element-wise tests. However, when I run the command `./bin/arm64/Release/ov_cpu_func_tests --gtest_filter="*smoke*Eltwise*"`, 76 tests failed. I'm not sure how to proceed further. Could you please guide me?

image

**Comment by eshoguli** [comment]
> Hey @eshoguli, I have incorporated the code review feedback and added the IsInf operation to the existing element-wise tests. However, when I run the command `./bin/arm64/Release/ov_cpu_func_tests --gtest_filter="*smoke*Eltwise*"`, 76 tests failed. I'm not sure how to proceed further. Could you please guide me?

The fix: https://github.com/inbasperu/openvino/commit/94e04bb6eeb818e9a7fb1fa15628b3e876ea79ea. I added the commit to your branch. Please, review the commit and rebase your branch to the latest master.

**Comment by eshoguli** [comment]
build_jenkins

File: src/tests/test_utils/common_test_utils/src/node_builders/eltwise.cpp [hunk 3]
```diff
@@ -12,6 +12,7 @@
 #include "openvino/op/divide.hpp"
 #include "openvino/op/erf.hpp"
 #include "openvino/op/floor_mod.hpp"
+#include "openvino/op/is_inf.hpp"
```
Comment (eshoguli) [inline src/tests/test_utils/common_test_utils/src/node_builders/eltwise.cpp:15]: minor: remove it, please.

**Review by eshoguli — APPROVED** [review]
Thanks! One minor comment, we are waiting CI completion

**Comment by dmitry-gorokhov** [comment]
build_jenkins

**Pull request merged**